nextcord/interactions.py

```diff
@@ -559,6 +539,485 @@ async def send(
+    application_id: :class:`int`
+        The application ID that the interaction was for.
+    user: Optional[Union[:class:`User`, :class:`Member`]]
+        The user or member that sent the interaction.
+    token: :class:`str`
+        The token to continue the interaction. These are valid
+        for 15 minutes.
+    data: :class:`dict`
+        The raw application command specific interaction data.
+    attached: :class:`InteractionAttached`
+        The attached data of the interaction. This is used to store any data you may need inside the interaction for convenience. This data will stay on the interaction, even after a :meth:`Interaction.application_command_before_invoke`.
+    application_command: Union[SlashApplicationSubcommand, BaseApplicationCommand]
+        The application command that triggered the interaction
+    app_command_name: :class:`str`
+        The name of the application command that triggered the interaction
+    app_command_id: :class:`int`
+        The application command ID that triggered the Interaction
+    options: List[SlashOptionData]
+        The options
+    """
+
+    __slots__: Tuple[str, ...] = (
+        "id",
+        "type",
+        "guild_id",
+        "channel_id",
+        "data",
+        "application_id",
+        "user",
+        "locale",
+        "guild_locale",
+        "token",
+        "version",
+        "attached",
+        "application_command",
+        "app_command_name",
+        "app_command_id",
+        "options",
+        "_permissions",
+        "_app_permissions",
+        "_state",
+        "_session",
+        "_original_message",
+        "_cs_response",
+        "_cs_followup",
+        "_cs_channel",
+    )
+
+    def __init__(self, *, data: InteractionPayload, state: ConnectionState):
+        super().__init__(data=data, state=state)
+
+        self.application_command: Optional[
+            Union[SlashApplicationSubcommand, BaseApplicationCommand]
+        ] = None
+
+    def _from_data(self, data: InteractionPayload):
+        super()._from_data(data=data)
+
+        self.app_command_name: str = self.data["name"]  # type: ignore # self.data should be present here
+        self.app_command_id: int = self.data["id"]  # type: ignore # self.data should be present here
+
+        try:
+            self.options: list = self._get_application_options(self.data["options"])  # type: ignore # self.data should be present here
+        except KeyError:
+            self.options: list = []
+
+    def _set_application_command(
+        self, app_cmd: Union[SlashApplicationSubcommand, BaseApplicationCommand]
+    ):
+        self.application_command = app_cmd
+
+    def _get_application_options(self, data):
+        options_collection: list = []
+
+        for option in data:
+            options_collection.append(SlashOptionData(option))
+
+        return options_collection
+
+
+class ApplicationAutocompleteInteraction(Interaction):
```

Comment:
If this class is going to copy a lot of logic from `ApplicationCommandInteraction`, shouldn't this class be a subclass of `ApplicationCommandInteraction`?